alibi/explainers/pd_variance.py

```diff
@@ -0,0 +1,660 @@
+            (assume hundreds/thousands of points) without providing any additional information for nearby points
+            (e.g., 0.35 and 351). By setting ``grid_resolution=5``, the marginal effect is computed for the values
+            ``[0.1, 0.3, 0.5, 0.7, 0.9]`` instead, which is less computationally demanding and can provide similar
+            insights regarding the model's behaviour. Note that the extreme values of the grid can be controlled
+            using the `percentiles` argument.
+        grid_points
+            Custom grid points. Must be a `dict` where the keys are the target features indices and the values are
+            monotonically increasing arrays defining the grid points for a numerical feature, and a subset of
+            categorical feature values for a categorical feature. If the `grid_points` are not specified,
+            then the grid will be constructed based on the unique target feature values available in the
+            dataset `X`, or based on the `grid_resolution` and `percentiles` (check `grid_resolution` to see when
+            it applies). For categorical features, the corresponding value in the `grid_points` can be
+            specified either as array of strings or array of integers corresponding the label encodings.
+            Note that the label encoding must match the ordering of the values provided in the `categorical_names`.
+
+        Returns
+        -------
+        explanation
+            An `Explanation` object containing the data and the metadata of the calculated partial dependence
+            curves and feature importance/interaction. See usage at `Partial dependence variance examples`_ for details
+
+            .. _Partial dependence variance examples:
+                https://docs.seldon.io/projects/alibi/en/stable/methods/PartialDependenceVariance.html
+        """
+        if method not in Method.__members__.values():
+            raise ValueError(f"Unknown mode. Received ``method={method}``. "
+                             f"Accepted `method` names are: {get_options_string(Method)}")
+
+        # get number of features
+        n_features = X.shape[1]
+
+        # construct features if not provided based on the method
+        if features is None:
+            features = list(range(n_features))
+
+            if method == Method.INTERACTION:
+                features = list(combinations(features, 2))
+
+        # compute partial dependence functions
+        params = {
+            'X': X,
+            'features': features,
+            'percentiles': percentiles,
+            'grid_resolution': grid_resolution,
+            'grid_points': grid_points
+        }
+
+        if not isinstance(self.pd_explainer.predictor, BaseEstimator):
+            params.update({'kind': Kind.AVERAGE.value})  # type: ignore[dict-item]
+
+        # compute partial dependence for each feature
+        pd_explanation = self.pd_explainer.explain(**params)  # type: ignore[arg-type]
```

Comment:
Interesting question on user experience - the `verbose` flag is passed through to the underlying explainer. I assume the calculation after we have `pd_explanation` is comparatively quick so it's fine to leave this as is?

Reply:
Needs to be addressed somehow. One way to do it is to add a new progress bar for the computation of the variance with an appropriate description. Unfortunately, the `PD` progress bar does not have any description. Not sure if it is best to add a description for the `PD` in this PR, or add a a progress bar without description for PDV and later, another PR to add a description for both.

Reply:
I don't mind where the PR's go too much. Would the proposal be to have 2 progress bars showing for this method or disabling the internal `PD` progress bar? Where is most of the time spent during the computation?

Reply:
The most time spent during computation is in the `PartialDependence`. The other is relatively fast. I proposal would be to have 2 progress bars for this method.

Reply:
If that's the case then perhaps easiest and simplest (for the user) to stick with the internal progress bar? Or perhaps we add the 2nd progress bar (with a different description), but disable the internal one?